### DISCUSSION TITLE
Valheim Backup

### DISCUSSION THREAD

**Discussion by patmann03** [GameServerManagers/LinuxGSM#3260] [Q&A] [answered]
I'm running a nightly backup using the bash file and a cron job.  I see its backing up the whole home directory (or most of it).  I was wondering if I just wanted to backup the game server itself would I just back up this folder: ~/.config/unity3d/IronGate/Valheim

The database (.db file) in the worlds folder seems to get updated every 30 minutes, is everything else held in memory until it writes to the db?  I had my brother logout and it didn't seem to trigger a database write.

Thanks in advance!

**Comment by blastermaster77** [comment]
why is lgsm not backing up the ~/.config/unity3d/IronGate/Valheim I now it loaded the world form backup and now I have a broken world :(

**Reply by patmann03** [reply]
Hmm, mine is.  I wonder if you have a permissions issue?

**Reply by EugeneGGHQ** [reply]
@blastermaster77 If you're still having problems, you could consider adding `-savedir config` to the server parameters into /home/vhserver/lgsm/config-lgsm/vhserver/vhserver.cfg so that it looks like:

`startparameters="-name '${servername}' -password ${serverpassword} -port ${port} -world ${gameworld} -public ${public} -savedir config"`

This will then create a new config directory with appropriate permissions (assuming that was the cause of the initial problem) to /home/vhserver/serverfiles/config

This method was mentioned in a [bug report](https://github.com/GameServerManagers/LinuxGSM/issues/3253) yesterday.

**Comment by EugeneGGHQ** [comment, marked as answer]
Though I haven't tested it yet, a safe bet would be to back up the home directory as you are currently including the config directory database too.

This would save you having to recreate your config files for server launch. You might be interested in using the '-savedir config' variable to [create a new server database under the serverfiles directory](https://github.com/GameServerManagers/LinuxGSM/discussions/3260#discussioncomment-373254).

**Reply by patmann03** [reply]
Thanks, that's what I setup with rsync.  Appreciate it.

**Reply by blastermaster77** [reply]
Thanks I modified it like this:

`#### Game Server Settings ####

servername="servername"
serverpassword="password"
port="2456"
gameworld="${selfname}"
public="1"
savedir="valheim_worlds"


## Server Parameters 
startparameters="-name '${servername}' -password ${serverpassword} -port ${port} -world ${gameworld} -public ${public} -savedir ${savedir}"`
